Failed IT on anvilbox https://gitlab.anvil.gi.ucsc.edu/ucsc/azul/-/jobs/82348

```
======================================================================
FAIL: test_indexing (integration_test.IndexingIntegrationTest.test_indexing)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/builds/ucsc/azul/test/integration_test.py", line 479, in test_indexing
    public_source, _ = self._select_source(
                       ~~~~~~~~~~~~~~~~~~~^
        catalog.name,
        ^^^^^^^^^^^^^
    ...<4 lines>...
        mirror=mirror and self._mirror_service(catalog.name).may_mirror()
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
  File "/builds/ucsc/azul/test/integration_test.py", line 340, in _select_source
    for source in self.managed_access_sources_by_catalog[catalog]
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/builds/ucsc/azul/src/azul/caching.py", line 203, in __get__
    value = obj.__dict__[self.fget.__name__] = self.fget(obj)
                                               ~~~~~~~~~^^^^^
  File "/builds/ucsc/azul/test/integration_test.py", line 287, in managed_access_sources_by_catalog
    all_sources = self._tdr_client.list_snapshots()
  File "/builds/ucsc/azul/src/azul/terra.py", line 603, in list_snapshots
    assert is_of_type(snapshot, self.Snapshot)
           ~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^
AssertionError
```

--- a/scripts/mirror.py
+++ b/scripts/mirror.py
@@ -29,7 +29,6 @@ def mirror_catalog(azul: AzulClient,
                    catalog: CatalogName,
                    source_globs: set[str],
                    wait: bool):
-    plugin = azul.repository_plugin(catalog)
     fail_queue = config.mirror_queue.to_fail.name
     assert azul.is_queue_empty(fail_queue), R(
         'Cannot begin mirroring because a previous operation failed: '
@@ -37,7 +36,7 @@ def mirror_catalog(azul: AzulClient,
         fail_queue)
     public_sources_by_spec = {
         source.spec: source
-        for source in plugin.list_sources(authentication=None)
+        for source in azul.source_service.list_sources(catalog, authentication=None)
     }
     # When the user doesn't specify a source or provides "*" as a source glob,
     # we implicitly filter out managed-access sources. This lets us assert that

--- a/scripts/post_deploy_tdr.py
+++ b/scripts/post_deploy_tdr.py
@@ -106,8 +106,8 @@ class TerraValidator:
             require(subgraph_count <= 512, 'Common prefix is too short', ref.spec)
 
     def verify_source_access(self) -> None:
-        public_snapshots = self.public_tdr.snapshot_ids()
-        all_snapshots = self.tdr.snapshot_ids()
+        public_snapshots = self.public_tdr.list_snapshot_ids()
+        all_snapshots = self.tdr.list_snapshot_ids()
         diff = public_snapshots - all_snapshots
         require(not diff,
                 'The public service account can access snapshots that the indexer '

--- a/src/azul/azulclient.py
+++ b/src/azul/azulclient.py
@@ -70,6 +70,9 @@ from azul.plugins import (
 from azul.queues import (
     Queues,
 )
+from azul.service.source_service import (
+    SourceService,
+)
 from azul.types import (
     JSON,
     JSONs,
@@ -107,6 +110,10 @@ class AzulClient(SignatureHelper, HasCachedHttpClient):
     @cache
     def mirror_service(self, catalog: CatalogName) -> BaseMirrorService:
         return BaseMirrorService(catalog=catalog)
+
+    @cached_property
+    def source_service(self) -> SourceService:
+        return SourceService()
 
     def local_reindex(self, catalog: CatalogName, prefix: str) -> int:
         service = self.index_repository_service

--- a/src/azul/plugins/repository/tdr.py
+++ b/src/azul/plugins/repository/tdr.py
@@ -139,7 +139,7 @@ class TDRPlugin[TDR_BUNDLE: TDRBundle,
                         authentication: Authentication | None
                         ) -> set[str]:
         def list_snapshot_ids(tdr: TDRClient):
-            return tdr.snapshot_ids()
+            return tdr.list_snapshot_ids()
 
         return self._auth_fallback(authentication, list_snapshot_ids)
 

--- a/src/azul/terra.py
+++ b/src/azul/terra.py
@@ -551,7 +551,7 @@ class TDRClient(SAMClient, DRSClient):
 
     page_size: ClassVar[int] = 1000
 
-    def snapshot_ids(self) -> set[str]:
+    def list_snapshot_ids(self) -> set[str]:
         """
         List the IDs of the TDR snapshots accessible to the current credentials.
         Much faster than listing the snapshots' names.

--- a/test/service/test_source_service.py
+++ b/test/service/test_source_service.py
@@ -211,7 +211,7 @@ class TestListSources(DCP2TestCase, LocalAppTestCase):
         _test(authenticate=False, cache=True)
         mock_source_service__get.return_value = None
         mock_source_service__get.side_effect = NotFound('foo_token')
-        with patch('azul.terra.TDRClient.snapshot_ids',
+        with patch('azul.terra.TDRClient.list_snapshot_ids',
                    return_value=self.source_names_by_id.keys() | {'not_indexed'}):
             _test(authenticate=True, cache=False)
             _test(authenticate=False, cache=False)

--- a/test/service/test_source_service.py
+++ b/test/service/test_source_service.py
@@ -140,22 +140,22 @@ class TestListSources(DCP2TestCase, LocalAppTestCase):
     def app_name(cls) -> str:
         return 'service'
 
-    source_names = ['mock_snapshot_1', 'mock_snapshot_2']
-    make_spec = 'tdr:bigquery:gcp:mock:{}'.format
+    snapshot_names = ['mock_snapshot_1', 'mock_snapshot_2']
+    make_spec_str = 'tdr:bigquery:gcp:mock:{}'.format
 
     # Includes extra sources to check that the endpoint only returns results
     # for the current catalog
     extra_sources = ['foo', 'bar']
-    source_names_by_id = {
+    snapshots_by_id = {
         str(i): name
-        for i, name in enumerate(source_names + extra_sources)
+        for i, name in enumerate(snapshot_names + extra_sources)
     }
 
     @classmethod
     def _sources(cls):
         return {
-            cls.make_spec(n): {'mirror': True}
-            for n in cls.source_names
+            cls.make_spec_str(n): {'mirror': True}
+            for n in cls.snapshot_names
         }
 
     @classmethod
@@ -172,9 +172,9 @@ class TestListSources(DCP2TestCase, LocalAppTestCase):
         return {
             cls.catalog: [
                 TDRSourceRef(id=id,
-                             spec=TDRSourceSpec.parse(cls.make_spec(name)),
+                             spec=TDRSourceSpec.parse(cls.make_spec_str(name)),
                              prefix=None)
-                for id, name in cls.source_names_by_id.items()
+                for id, name in cls.snapshots_by_id.items()
                 if name not in cls.extra_sources
             ]}
 
@@ -182,7 +182,7 @@ class TestListSources(DCP2TestCase, LocalAppTestCase):
     @patch.object(TDRClient, 'list_snapshots')
     @patch.object(TDRClient, 'validate', new=MagicMock())
     def test(self, mock_tdr_client__list_snapshots, mock_source_service__get):
-        mock_tdr_client__list_snapshots.return_value = self.source_names_by_id
+        mock_tdr_client__list_snapshots.return_value = self.snapshots_by_id
         client = http_client(log)
         azul_url = furl(url=self.base_url,
                         path='/repository/sources',
@@ -198,20 +198,20 @@ class TestListSources(DCP2TestCase, LocalAppTestCase):
                     'sources': [
                         {
                             'sourceId': id,
-                            'sourceSpec': str(TDRSourceSpec.parse(self.make_spec(name)))
+                            'sourceSpec': str(TDRSourceSpec.parse(self.make_spec_str(name)))
                         }
-                        for id, name in self.source_names_by_id.items()
+                        for id, name in self.snapshots_by_id.items()
                         if name not in self.extra_sources
                     ]
                 }
                 self.assertEqual(expected, actual)
 
-        mock_source_service__get.return_value = list(self.source_names_by_id.keys())
+        mock_source_service__get.return_value = list(self.snapshots_by_id.keys())
         _test(authenticate=True, cache=True)
         _test(authenticate=False, cache=True)
         mock_source_service__get.return_value = None
         mock_source_service__get.side_effect = NotFound('foo_token')
         with patch('azul.terra.TDRClient.list_snapshot_ids',
-                   return_value=self.source_names_by_id.keys() | {'not_indexed'}):
+                   return_value=self.snapshots_by_id.keys() | {'not_indexed'}):
             _test(authenticate=True, cache=False)
             _test(authenticate=False, cache=False)

